### PR TITLE
(#2) Test to Make Sure Version Never Published

### LINKED ISSUE #2: Story: Never been released before check for NPM
## As a gatekeeper, I want to make sure that upon push to main, the package version is checked to be sure that version of code has never been released, so that we do not allow developers to try to publish previously released versions.

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: Developer submits code with the same package version as a previous release
  Given a new commit is pushed to main
    And the package.json version is 1.0.0
    And there exists a tag named `v1.0.0`
  When  github actions runs cgov-version-check-action
  Then  an error message with `The following checks have failed: hasNeverBeenReleased` appears in the run logs
    And the step fails

```

### Notes
* The check's name should be `hasNeverBeenReleased`.

### PR DESCRIPTION
**Pull request by bryanpizzillo**
This has been pushed as a pre-release to v0.0.8-beta1.

Closes #2

### CODE REVIEW

**Comment by bryanpizzillo** [comment]
I tested v0.0.8-beta1:
* When there were 0 releases - should pass and did pass
* When there was a release with the same version - should fail and did fail
* When there were releases, and this version has not been published - should pass and did pass.

https://github.com/bryanpizzillo/package-release-test/actions

File: src/lib/npm/__tests__/has-never-been-released.test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		getPackageVersion.mockReturnValueOnce('1.2.2');
+
+		// Mock the response to get all released versions
+		getAllGithubReleases.mockReturnValueOnce([]);
+
+		const actual = await hasNeverBeenReleased(mockOctokit, mockContext);
+
+		expect(getPackageVersion).toHaveBeenCalledWith(
+			mockOctokit,
+			'userName',
```
Comment (blairlearn) [inline src/lib/npm/__tests__/has-never-been-released.test.js:34]: There's a wrinkle here. Although you know the correct value was passed, you don't know whether it's the correct _instance_ of that value.

I tracked down the value `'userName'` to `mock-push-payload.json` and discovered the authors, committers, pusher, and account owner are all the same person.

What I'm not certain of is whether changing that is really worth the level of effort.

File: src/lib/npm/__tests__/has-never-been-released.test.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+		expect(getAllGithubReleases).toHaveBeenCalledWith(
+			mockOctokit,
+			'userName',
+			'package-release-test'
+		);
+
+		expect(actual).not.toBeTruthy();
+	});
+	// TODO: Handle error from api
```
Comment (blairlearn) [inline src/lib/npm/__tests__/has-never-been-released.test.js:87]: Is this still outstanding?

Comment (bryanpizzillo) [reply src/lib/npm/__tests__/has-never-been-released.test.js:87]: Yep. Guess I gotta go implement that...

Comment (bryanpizzillo) [reply src/lib/npm/__tests__/has-never-been-released.test.js]: Added. Guess that is what happens when weeks go by...

File: src/lib/npm/__tests__/has-never-been-released.test.js (hunk 1 above)
Comment (bryanpizzillo) [reply src/lib/npm/__tests__/has-never-been-released.test.js:34]: Ok, I am going to leave it as  that content **WAS** the same for an actual request.

**Review by blairlearn — APPROVED** [review]

**Pull request merged**